### PR TITLE
[fix]: [PL-22214]: fix for successful entities getting queued on retrigger

### PR DESCRIPTION
**Pull request by bhavya181**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)

### CODE REVIEW

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 2 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityService.java, line 36 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-Muk6bls8AWaNIFx-Lgl:-Muk6bls8AWaNIFx-Lgm:b-m572eo) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityService.java#L36)):*
> ```Java
>       GitFullSyncEntityInfoFilterDTO gitFullSyncEntityInfoFilterDTO);
> 
>   List<GitFullSyncEntityInfo> listEntitiesToBeSynced(String accountIdentifier, String fullSyncJobId);
> ```


Can you check that this function is called from only one place ?

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncProcessorServiceImpl.java, line 80 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-Muk6mJd697oL51be17A:-Muk6mJd697oL51be17B:b8x8wcp) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncProcessorServiceImpl.java#L80)):*
> ```Java
>       fullSyncResponse = performSyncForEntities(microservice, entityInfoList);
>     } catch (Exception e) {
>       failed = true;
> ```

lets name it requestFailed

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/repositories/fullSync/GitFullSyncEntityRepositoryCustom.java, line 31 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-Muk6x2ODQpqaDr_X1SS:-Muk6x2ODQpqaDr_X1ST:b-dalvyy) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/repositories/fullSync/GitFullSyncEntityRepositoryCustom.java#L31)):*
> ```Java
> 
>   List<GitFullSyncEntityInfo> findAll(Criteria criteria);
> }
> ```

Why did we create a new function ?

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 6 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityServiceImpl.java, line 72 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-MukGyNp2MoELB3dGEJE:-MukGyNp2MoELB3dGEJF:bia5ahu) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityServiceImpl.java#L72)):*
> ```Java
> 
>   @Override
>   public List<GitFullSyncEntityInfo> listEntitiesToBeSynced(String accountIdentifier, String messageId) {
> ```

Can it create an issue if the returned list had an entity that was in QUEUED state, but before the next step happened, it turned to FAILURE/SUCCESS?

It is even possible?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 7 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityService.java, line 36 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-Muk6bls8AWaNIFx-Lgl:-MukZf9j3RdcsqW_5Ull:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityService.java#L36)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Can you check that this function is called from only one place ?
</blockquote></details>

Done.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityServiceImpl.java, line 72 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-MukGyNp2MoELB3dGEJE:-MukZtHW8hpilryXr7Xn:b-uva2iw) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityServiceImpl.java#L72)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Can it create an issue if the returned list had an entity that was in QUEUED state, but before the next step happened, it turned to FAILURE/SUCCESS?

It is even possible?
</blockquote></details>

No, resync will be triggered only when entities are failed, no processing of entities will occur for earlier sync, such race condition will not occur.
___
*[136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncProcessorServiceImpl.java, line 80 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-Muk6mJd697oL51be17A:-MukZhBv0V99m3yH4GW8:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncProcessorServiceImpl.java#L80)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

lets name it requestFailed
</blockquote></details>

Done
___
*[136-git-sync-manager/src/main/java/io/harness/repositories/fullSync/GitFullSyncEntityRepositoryCustom.java, line 31 at r1](https://reviewable.io/reviews/harness/harness-core/30385#-Muk6x2ODQpqaDr_X1SS:-MukZkIwB1VmjSTA4fk_:brwvm0t) ([raw file](https://github.com/harness/harness-core/blob/e6cd6c8bb1dfd466bfd7a723e28922fc0ffa6880/136-git-sync-manager/src/main/java/io/harness/repositories/fullSync/GitFullSyncEntityRepositoryCustom.java#L31)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Why did we create a new function ?
</blockquote></details>

there is no method to do findAll  with query.

**Review by DeepakPatankar — COMMENTED** [review]
> Reviewed 2 of 6 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 7 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385)* status: 7 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by bhavya181** [comment]
trigger messagecheck

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityServiceImpl.java [hunk 1]
```diff
@@ -69,8 +69,14 @@ public void markSuccessful(String uuid, String accountId) {
...
-  public List<GitFullSyncEntityInfo> list(String accountIdentifier, String messageId) {
-    return gitFullSyncEntityRepository.findByAccountIdentifierAndMessageId(accountIdentifier, messageId);
+  public List<GitFullSyncEntityInfo> listEntitiesToBeSynced(String accountIdentifier, String messageId) {
+    Criteria criteria = Criteria.where(GitFullSyncEntityInfoKeys.accountIdentifier)
+                            .is(accountIdentifier)
+                            .and(GitFullSyncEntityInfoKeys.messageId)
+                            .is(messageId)
+                            .and(GitFullSyncEntityInfoKeys.syncStatus)
+                            .in(Arrays.asList(QUEUED, FAILED));
+    return gitFullSyncEntityRepository.findAll(criteria);
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/core/fullsync/GitFullSyncEntityServiceImpl.java:79]: the list should be defined somewhere

**Comment by DeepakPatankar** [comment]
trigger messagecheck

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 5 of 6 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30385#-:-MulWJak5wazNDrpiBxe:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**